src/Network/Network/AzureFirewallPolicy/ApplicationRuleCondition/NewAzureFirewallPolicyApplicationRuleCommand.cs

```diff
@@ -0,0 +1,94 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Network.Models;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "FirewallPolicyApplicationRule", SupportsShouldProcess = true, DefaultParameterSetName = AzureFirewallPolicyApplicationRuleParameterSets.TargetFqdn), OutputType(typeof(PSAzureFirewallPolicyApplicationRule))]
+    public class NewAzFirewallPolicyApplicationRuleCommand : AzureFirewallPolicyBaseCmdlet
+    {
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The name of the Application Rule")]
+        [ValidateNotNullOrEmpty]
+        public virtual string Name { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The description of the rule")]
+        [ValidateNotNullOrEmpty]
+        public string Description { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The source addresses of the rule")]
+        [ValidateNotNullOrEmpty]
+        public string[] SourceAddress { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = AzureFirewallPolicyApplicationRuleParameterSets.TargetFqdn,
+            HelpMessage = "The target FQDNs of the rule")]
+        [ValidateNotNullOrEmpty]
+        public string[] TargetFqdn { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = AzureFirewallPolicyApplicationRuleParameterSets.FqdnTag,
+            HelpMessage = "The FQDN Tags of the rule")]
+        [ValidateNotNullOrEmpty]
+        public string[] FqdnTag { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = AzureFirewallPolicyApplicationRuleParameterSets.TargetFqdn,
+            HelpMessage = "The protocols of the rule")]
+        [ValidateNotNullOrEmpty]
+        public string[] Protocol { get; set; }
```

Comment:
No we do not have a  set of known values.This takes the format http:80 for an example. And as we do not have a limited number of ports, the port:protocol combination is not a limited set.

Reply:
Its because we do not have a single protocol and a single port. we have a list of port:protocol combination. If we separate them out we cannot correlate one protocol to one port.

Also this current format is already being followed by firewall and wanted to keep it consitent. Also we have the same format for the firewallpolicy portal UI and the user is familiar with this format.We have not faced any issues with this before. We will also explicitly mention this in the documentation.

Reply:
Cool. Thanks for sharing that information!